docs/user-guide/src/tour/new.md

```diff
@@ -248,66 +102,54 @@ error: [Prusti: invalid specification] use of impure function "Link::len" in pur
 
 Whenever we add the attribute `#[pure]` to a function, Prusti will check whether that
 function is indeed deterministic and side-effect free
-(notice that termination is *not* checked); otherwise, it complains.
+(notice that [termination](../capabilities/limitations.md#termination-checks-total-correctness-missing) is *not* checked); otherwise, it complains.
 In this case, Prusti complains because we call an impure function,
 namely `Link::len()`, within the body of the pure function `List::len()`.
 
 To fix this issue, it suffices to mark `Link::len()` as pure as well.
 
 ```rust,noplaypen
-#//uncomment: #![feature(box_patterns)]
-#use prusti_contracts::*; 
-#
-#pub struct List {
-#    head: Link,
-#}
-#
-#enum Link {
-#    Empty,
-#    More(Box<Node>),
-#}
-#
-#struct Node {
-#    elem: i32,
-#    next: Link,
-#}
-#
-#impl List {
-#    #[pure]
-#    pub fn len(&self) -> usize {
-#        self.head.len()
-#    }
-#
-#    #[ensures(result.len() == 0)]
-#    pub fn new() -> Self {
-#        List {
-#            head: Link::Empty,
-#        }
-#    }
-#}
-#
-impl Link {
-    #[pure]
-    fn len(&self) -> usize {
-        0
-    }
-}
-#
-# fn main() {} // in case Prusti is used via command line
-#
+{{#rustdoc_include ../../../../prusti-tests/tests/verify/pass/user-guide/impl_new_spec_fixed.rs:pure_annotation}}
 // Prusti: VERIFIES
 ```
 
-```
-$ prusti-rustc list.rs
+```plain
+$ cargo-prusti
 // ...
 Successful verification of 4 items
 ```
 
 Prusti now manages to verify that `new()` always returns
 a list for which the method `len()` returns 0. (notice
-this is hardly surprising since `len()` ultimately always returns 0 but we will change
-this soon.)
+this is hardly surprising since `len()` ultimately always returns 0.)
+
+## Proper implementation of `len`
+
+We will now properly implement `len()`, and while we're at it, `is_empty()` for `Link`. Both of them are pure functions, so we will add the `#[pure]` annotation. Both functions can be called without any restrictions, so they have the default postcondition `#[requires(true)]`, which we don't have to add manually. We also don't need to add any additional postconditions, since pure functions will be inlined wherever they are used during verification.
```

Comment:
"inlined wherever they are used during verification" is not completely accurate. This might suggest recursion is not possible, for example. You should rather say that the body of the pure function is implicitly part of its specification.